engine/src/main/java/org/camunda/bpm/engine/impl/cfg/CompositeProcessEnginePlugin.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.impl.cfg;
+
+
+import org.camunda.bpm.engine.ProcessEngine;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * {@link ProcessEnginePlugin} that provides composite behavior. When registered on an engine configuration,
+ * all plugins added to this composite will be triggered on preInit/postInit/postProcessEngineBuild.
+ * <p>
+ * Use to encapsulate common behavior (like engine configuration).
+ */
+public class CompositeProcessEnginePlugin extends AbstractProcessEnginePlugin {
+
+  protected final List<ProcessEnginePlugin> plugins;
+
+  /**
+   * New instance with empty list.
+   */
+  public CompositeProcessEnginePlugin() {
+    this.plugins = new ArrayList<ProcessEnginePlugin>();
+  }
+
+
+  /**
+   * New instance with vararg.
+   * @param plugin first plugin
+   * @param additionalPlugins additional vararg plugins
+   */
+  public CompositeProcessEnginePlugin(ProcessEnginePlugin plugin, ProcessEnginePlugin... additionalPlugins) {
```

Comment:
if you do have a no-arg public constructor, is there really a point in forcing "plugin" parameter?